packages/react-native-worklets/src/threads.ts

```diff
@@ -77,11 +78,53 @@ export const callMicrotasks = SHOULD_BE_USE_WEB
  * @param args - Arguments to pass to the function.
  * @see https://docs.swmansion.com/react-native-worklets/docs/threading/scheduleOnUI
  */
+// @ts-expect-error This overload is correct since it's what user sees in his code
+// before it's transformed by Worklets Babel plugin.
 export function scheduleOnUI<Args extends unknown[], ReturnValue>(
   worklet: (...args: Args) => ReturnValue,
   ...args: Args
+): void;
+export function scheduleOnUI<Args extends unknown[], ReturnValue>(
```

Comment:
```suggestion
): void;

export function scheduleOnUI<Args extends unknown[], ReturnValue>(
```

Maybe add a line break for readability?